### DISCUSSION TITLE
Diskussion: Tagesstatistiken

### DISCUSSION THREAD

**Discussion by ArchGithub** [nobl/ioBroker.senec#507] [General]
I have linked the data and found that the Stata_Day data is not working!

![image](https://user-images.githubusercontent.com/38007195/89130851-c7659780-d508-11ea-880e-909acb1326e7.png)

**Comment by nobl** [comment]
I suppose your Senec was updated lately?
STAT_DAY was removed from the api by Senec a while ago unfortunately.

I would like to suggest using javascript adapter to maybe take care of generating day-stats. You would just have to use the total counter for any of the values at the beginning of the day and then updating that.

See also #2 and #4

**Comment by Pr1smBr34k3r** [comment]
Hello,
i've solved it with the following solution:

**1. Add the following objects under the javascript adapter:**
![Senec](https://user-images.githubusercontent.com/12789388/89203568-110bbc00-d5b5-11ea-9b21-04d9258cd613.png)

**2. Create the following Script with blockly:**

![blockly1](https://user-images.githubusercontent.com/12789388/89205161-8ed0c700-d5b7-11ea-80f0-59d8f8cb0451.png)


**Please consider that some values i have posted are not needed for your day statistic. Remove it, if you not need it.**

**You need this block for the daily statistic:**
![blockly2](https://user-images.githubusercontent.com/12789388/89204649-bc694080-d5b6-11ea-8fd4-515b9cc703fe.png)


**This is for monthly stats:**
![blockly3](https://user-images.githubusercontent.com/12789388/89205188-9db77980-d5b7-11ea-93e4-ea02e3e58b14.png)


I'm hope it helps you.

Greetz ZeR0

**Comment by ArchGithub** [comment]
> Hello,
> i've solved it with the following solution:
> 
> **1. Add the following objects under the javascript adapter:**
> ![Senec](https://user-images.githubusercontent.com/12789388/89203568-110bbc00-d5b5-11ea-9b21-04d9258cd613.png)
> 
> **2. Create the following Script with blockly:**
> 
> ![blockly1](https://user-images.githubusercontent.com/12789388/89205161-8ed0c700-d5b7-11ea-80f0-59d8f8cb0451.png)
> 
> **Please consider that some values i have posted are not needed for your day statistic. Remove it, if you not need it.**
> 
> **You need this block for the daily statistic:**
> ![blockly2](https://user-images.githubusercontent.com/12789388/89204649-bc694080-d5b6-11ea-8fd4-515b9cc703fe.png)
> 
> **This is for monthly stats:**
> ![blockly3](https://user-images.githubusercontent.com/12789388/89205188-9db77980-d5b7-11ea-93e4-ea02e3e58b14.png)
> 
> I'm hope it helps you.
> 
> Greetz ZeR0


Hello,

Thanks for the info.
I'll try it out sometime these days.
Gesamt_Heute is then Gesamt_Gestern or?
Something is wrong with your feed-in Einspeisungswerten, because the value of the aktueller_Tagesstand is higher than the Gesamt_heute?

![image](https://user-images.githubusercontent.com/38007195/89270445-a5b8fd00-d63b-11ea-8269-03312e7fd20d.png)

Another question is, do you know how often these values are updated in memory.

![image](https://user-images.githubusercontent.com/38007195/89270716-0ba58480-d63c-11ea-9c80-3f2762c69611.png)

I also find the other values interesting.
So how did you do that with the calculation of the feed-in tariff?

Danke!

Many greetings

**Comment by Pr1smBr34k3r** [comment]
> Gesamt_Heute is then Gesamt_Gestern or?
Something is wrong with your feed-in Einspeisungswerten, because the value of the aktueller_Tagesstand is higher than the Gesamt_heute?

Yes you are right. 
I have missnamed some values. 
***_Heute (Today) are the Values ***_Gestern (Yesterday).

But for me, this values are generated at 23:55 and saved in a sql database. So these are the Sum of the actual day. 
The values ***_aktueller_Tagesstand are the live view for the day from 00:00 - 23:55 and get stored in ***_Heute at the end of the day at 23:55. Then ***_aktueller_Tagesstand beginns fresh from 0.


> I also find the other values interesting.
So how did you do that with the calculation of the feed-in tariff?

What exactly do you want to know about the feed-in tariff?
I would like to explain it to you. But don't understand what you want to know exactly? Are you missing the values that I used in the script? Im using the SENEC.Home V3 hybrid duo. Possibly older Senec (like V2) has different values. @nobl can perhaps say a little bit more about this.

**Comment by Pr1smBr34k3r** [comment]
> Another question is, do you know how often these values are updated in memory.

I've forget to answer this.
Senec.Einspeisung.Aktueller_Tagesstand get updatet every 5sec. (see the script */5 * * * *) you can also set it shorter.
Senec.Einspeisung.Gesamt_Heute get updatet every day at 23:55 (see the script 55 23 * * *)

This may help you to set the time:
```
┌───────────── minute (0 - 59)
│ ┌───────────── hour (0 - 23)
│ │ ┌───────────── day of the month (1 - 31)
│ │ │ ┌───────────── month (1 - 12)
│ │ │ │ ┌───────────── day of the week (0 - 6) (Sunday to Saturday;
│ │ │ │ │                                   7 is also Sunday on some systems)
│ │ │ │ │
│ │ │ │ │
* * * * * 
```

**Comment by ArchGithub** [comment]
> > Another question is, do you know how often these values are updated in memory.
> 
> I've forget to answer this.
> Senec.Einspeisung.Aktueller_Tagesstand get updatet every 5sec. (see the script */5 * * * *) you can also set it shorter.
> Senec.Einspeisung.Gesamt_Heute get updatet every day at 23:55 (see the script 55 23 * * *)
> 
> This may help you to set the time:
> 
> ```
> ┌───────────── minute (0 - 59)
> │ ┌───────────── hour (0 - 23)
> │ │ ┌───────────── day of the month (1 - 31)
> │ │ │ ┌───────────── month (1 - 12)
> │ │ │ │ ┌───────────── day of the week (0 - 6) (Sunday to Saturday;
> │ │ │ │ │                                   7 is also Sunday on some systems)
> │ │ │ │ │
> │ │ │ │ │
> * * * * * 
> ```

Danke, ich habe es erfolgreich umgesetzt!

**Comment by Sebastian70-sys** [comment]
Hi,
I think your article is great. I also have a Senec memory. I just can't get it to work.
I am new to iobroker. Can you maybe send me a screenshot of your Javascript adapter.
Where the scripts and blockly are placed.
It would help me a lot.

Many greetings

**Comment by nobl** [comment]
Are you having issues with the Adapter (use latest repository, install and configure it - done). Just like with every other adapter.

You can access the adapter's datapoints via blockly or whatever you prefer.

**Comment by Sebastian70-sys** [comment]
Hi, thanks for the quick response. I'm too stupid to set that up :-). Hence my question about the screen sheet. It looks like this for me. As I said, I don't know my way around very well.
![script](https://user-images.githubusercontent.com/79689197/109288661-765f1d00-7825-11eb-8e2b-c3b7a6a8aa27.JPG)
![script2](https://user-images.githubusercontent.com/79689197/109288666-7828e080-7825-11eb-9bca-1a171ec00e86.JPG)

**Comment by nobl** [comment]
That is not where you install adapters. 

Installing is just hitting install - no more than that. (You just need to switch to latest Repository in iobroker settings.): 
![grafik](https://user-images.githubusercontent.com/4102132/109294386-63e8e180-782d-11eb-86d3-36a2d22eed1c.png)
followed by configuring it:
![grafik](https://user-images.githubusercontent.com/4102132/109294452-80851980-782d-11eb-9990-bba00a4ad129.png)

If you are absolutely new to iobroker, I would like to suggest checking out the homepage and forum: https://forum.iobroker.net

What @git-ZeR0 and @ArchGithub describe above is only related to the adapter and describes how to calculate some special data using Blockly.

(Just in case: German would be ok for me, too.)

**Comment by Sebastian70-sys** [comment]
Super Danke.
Den Adapter hab ich jetzt Aktuallisiert und er läuft. Der ist echt Klasse. So viele Informationen. 
Mir ging es auch darum die Tageswerte zu berechnen damit ich sie in der Vis anzeigen kann.

**Comment by nobl** [comment]
Die Tageswerte kannst Du eigentl. 1:1 mit Blockly nachbauen, wie von den beiden oben beschrieben.
Einfach die entspr. Datenpunkte erstmal anlegen und dann alles mit Blockly nachbauen. Wenn du damit aktuell noch nicht klarkommst, würde ich allerdings empfehlen, dass du dich da vielleicht mal im Forum etwas umsiehst, wie man mit Blockly arbeitet.

Das ist zwar eigentl. "straight forward" aber trotzdem nicht ganz ohne Fallstricke - gerade dann, wenn man noch nicht damit gearbeitet hat. Da ist z.B. auch die Farbe der Bausteine wichtig, um den korrekten auszuwählen.

**Comment by hefe193** [comment]
Hallo, ich bin neu hier ;)
Danke für den tollen Adapter, der mit meinem SENEC Home V3 hybrid duo gut funktioniert.
Nun mal direkt meine Frage(n):
Die Tageswerteberechnung basiert aber zum Teil auf den Daten mit niedriger Priorität des Adapters oder?
Warum ist da die minimale Abfrage auf 60min begrenzt?

So aktualisieren sich die Tageswerte doch erst alle Stunde oder?

Frage mich warum dann die Tageswerte alle 5 sec neu berechnet werden.
Oder: ich hab es vielleicht nicht verstanden.

Viele Grüße
Henning

**Comment by nobl** [comment]
Moin!

Die Berechnung der Tageswerte alle x Sekunden oder x Minuten ist eigentl. recht egal.
Ich persönlich würde diese sogar erst dann berechnen lassen, wenn sich der zugehörige Datenkpunkt ändert oder aktualisiert. Damit wären die berechneten Daten immer so aktuell wie die zugehörigen Datenpunkte. Das ist aber eine individuelle Sache.

Theoretisch könnte man den Tageswert auch erst irgendwann um 22 Uhr berechnen lassen, wenn keine Sonne mehr scheint.
Nötig ist das ganze ja leider nur, weil Senec die Tagesdaten aus der API genommen hat mir neueren Versionen (ich bekomme noch die Tageswerte).
Genaugenommen hole ich mir meine Werte der PV Produktion sogar überhaupt nicht mehr vom Schätzeisen "Senec", sondern direkt vom Wechselrichter.

Für den Rest (insbesondere auch wegen der Aufteilung Low / High Prio) kopiere ich gerade mal meine Antwort aus dem ioBroker Forum :) 
Eine unbedachte Manipulation kann da nämlich unschöne Konsequenzen haben.

Aktuell sind folgende Daten High-Prio:

'"ENERGY":{"STAT_STATE":"","STAT_STATE_DECODE":"","GUI_BAT_DATA_POWER":"","GUI_INVERTER_POWER":"","GUI_HOUSE_POW":"","GUI_GRID_POW":"","**GUI_BAT_DATA_FUEL_CHARGE**":"","GUI_CHARGING_INFO":"","GUI_BOOSTING_INFO":"","GUI_BAT_DATA_POWER":"","GUI_BAT_DATA_VOLTAGE":"","GUI_BAT_DATA_CURRENT":"","GUI_BAT_DATA_FUEL_CHARGE":"","GUI_BAT_DATA_OA_CHARGING":"","STAT_LIMITED_NET_SKEW":""}';

"PV1":{"POWER_RATIO":"","MPP_POWER":"","MPP_INT":""}';
"PWR_UNIT":{"POWER_L1":"","POWER_L2":"","POWER_L3":""}';
"PM1OBJ1":{"FREQ":"","U_AC":"","I_AC":"","P_AC":"","P_TOTAL":""}';
"PM1OBJ2":{"FREQ":"","U_AC":"","I_AC":"","P_AC":"","P_TOTAL":""}';

Genaugenommen ist im HighPrio Bereich der Großteil (wenn nicht sogar alles oder mehr) dessen, was auch über die lokale GUI eingesehen werden kann (wobei diese ja z.B. keinen Füllstand anzeigt).

Alles andere ist LowPrio.
(Da sind mir folgende Gruppen aktuell bekannt: '{"STATISTIC":{},"ENERGY":{},"FEATURES":{},"LOG":{},"SYS_UPDATE":{},"WIZARD":{},"BMS":{},"BAT1":{},"BAT1OBJ1":{},"BAT1OBJ2":{},"BAT1OBJ2":{},"BAT1OBJ3":{},"BAT1OBJ4":{},"PWR_UNIT":{},"PV1":{},"FACTORY":{},"GRIDCONFIG":{}}';)

Gefühlt kann es sein, dass einem z.B. GUI_BAT_DATA_FUEL_CHARGE als LowPrio vorkommt, da dieser Datenpunkt nur aktualisiert wird, wenn es auch etwas zu aktualisieren gibt (und das ist möglicherweise erst nach Minuten oder länger der Fall).

Eine individuelle Anpassung Low/High wollte ich schon seit geraumer Zeit einbauen - das ist gerade aber eher ein Zeitproblem.

Allerdings (daher auch die eindringliche Warnung in der Hilfe):
Werden zuviele Daten in zu kurzer Zeit abgefragt, dann verweigert die Senec ihren Dienst (keinerlei Verbindung mehr zur Plattform, was Garantieprobleme nach sich ziehen kann) und das kann sogar bis hin zum ungeplanten Neustart gehen.

Genau aus diesem Grund habe ich mich bei HighPrio auf die wichtigsten Datenpunkte beschränkt und bin äußerst zurückhaltend, bei der schnelleren Abfrage von weiteren Datenpunkten.

Theoretisch könntest Du in der Source einfach die entspr. Datenpunkte bei HighPrio mit reinschreiben und gut - zumindest so lange, bis ich es geschafft habe, dass konfigurierbar zu machen (da muss man dann aber auch genau wissen, was man angibt).
Allerdings der wichtige Hinweis: Ich bin nicht dafür verantwortlich, sollte es dadurch Probleme mit der Anlage oder sogar zu Problemen mit der Garantie kommen, weshalb ich ein solches Vorgehen keinesfalls empfehlen kann.

**Comment by hefe193** [comment]
Hi, danke für deine Antwort!

Mir geht es auch nicht um eine häufigere Datensammlung, auch die high priority reicht eigentlich alle 30 oder sogar 60s. Alles besser als die 5 Minuten der Senec App inkl. Serverausfall.

Ich wollte mir mit den Daten sowas wie die Übersicht von dem jung Viso pro Server nachbauen. Da haben sie die Tageswerte ja unterhalb der aktuellen stehen. Dient also eigentlich nur der Befriedigung der Grünen Seele und zum angeben bei Freunden die mit kWh sicher mehr anfangen als mit kW. Die aktuellen Werte sollen dann aber wenn ich im Sommer doch häufiger in die Begrenzung falle dazu herhalten einen Heizstab zu schalten um die WP im Bereich warm Wasser zu entlasten.

Viele Grüße
Henning Feddeck

**Comment by nobl** [comment]
Zur Information:
Die nächste Version wird einen eigenen Channel haben, um Tageswerte direkt im Adapter zu berechnen.

Dabei handelt es sich um die am häufigsten vermissten Werte in aktuelleren Versionen der SENEC:
Erzeugung, Verbrauch, Gespeichert, Entladen, Bezug, Einspeisung

Wer neugierig ist und schonmal testen will:
Den Adapter einfach über Github updaten - dort liegt bereits die aktuelle Version. Also unter "Adapters" über diesen Button:
![grafik](https://user-images.githubusercontent.com/4102132/116818096-dc3f8c00-ab69-11eb-9e49-b0f539219d16.png)

Die Werte aktualisieren sich nur bei Bedarf. Sprich dann, wenn sich der zugrundeliegende Basiswert ebenfalls geändert hat.

**Comment by Tw1nh34d** [comment]
> 
> 
> Hello,
> i've solved it with the following solution:
> 
> **1. Add the following objects under the javascript adapter:**
> ![Senec](https://user-images.githubusercontent.com/12789388/89203568-110bbc00-d5b5-11ea-9b21-04d9258cd613.png)
> 
> **2. Create the following Script with blockly:**
> 
> ![blockly1](https://user-images.githubusercontent.com/12789388/89205161-8ed0c700-d5b7-11ea-80f0-59d8f8cb0451.png)
> 
> **Please consider that some values i have posted are not needed for your day statistic. Remove it, if you not need it.**
> 
> **You need this block for the daily statistic:**
> ![blockly2](https://user-images.githubusercontent.com/12789388/89204649-bc694080-d5b6-11ea-8fd4-515b9cc703fe.png)
> 
> **This is for monthly stats:**
> ![blockly3](https://user-images.githubusercontent.com/12789388/89205188-9db77980-d5b7-11ea-93e4-ea02e3e58b14.png)
> 
> I'm hope it helps you.
> 
> Greetz ZeR0

Hi ZeR0,

könntest du mir/hier auch deine Variablen und Bockly Scripte für Senec.Akku.* & Senec.Strom.* veröffentlichen? uind evtl noch weiteres Nützliches zu diesem Thema :)
Ich würde mich sehr darüber freuen!!!

Viele Grüße

**Comment by Pr1smBr34k3r** [comment]
Hi Tw1nh34d,

ich denke das Thema sollte sich mit der nächsten Version des Adapters eh weitgehend erledigt haben, da NoBl die Berechnung der Werte in den Adapter integriert hat. Somit entfällt ein Großteil, wenn nicht sogar alles der Berechnungen über Blockly. Senec Akku, z.B. berechne ich gar keine Ladung und Entladung. Sondern nur die Laufzeit, wie lange der Akku noch hält oder braucht zum vollständigen Aufladen. Im neuen Adapter ist das aber drin. Was meinst du mit Senec.Strom?

Btw. der Adapter ist schon in Github als Vorabversion drin und kann auch von dort bereits installiert und getestet werden. Je mehr Leute testen und ein Feedback geben, desto eher wird NoBl die Freigabe geben. Vermute ich ;)

**Comment by Tw1nh34d** [comment]
hey, danke für deine rasche Antwort.
Ich habe es gerade gesehen, die Daten stehen unter _calc. Für die Monatsberechnung brauchts dann doch noch z.b., deine Blocklies :)
Bzgl. Senec.Strom, du berechnest scheinbar Bezugskosten, Kostenersparnis etc. Darauf hatte ich mich eigentlich bezogen. Hast du hierfür weitere Scripte erstellt?

**Comment by Pr1smBr34k3r** [comment]
Ja gut die "Kosten Ersparnis" ist jetzt kein Hexenwerk zu berechnen, zumindest so wie ich es derzeit habe. Das kann man so eh nicht genau nehmen. Ich rechne hier nur PV-Erzeugung-Total minus Export, das ist ja dann was ich wirklich derzeit von meinem eigen prodiziertem Strom selbst verwendet habe und das x Strompreis pro kWh + Export x Exportpreis. Was ja so aber eigentlich nicht korrekt ist. Dazu müsste man dann noch die Steuer berücksichtigen. Konnte das nur bis heute noch nie richtig fertig stellen, da ich nach nun gut über einem Jahr immer noch nicht mal einen Einspeisungvertrag habe und somit auch nicht weiss, was ich Vergütung bekomme. Die Monatsberechnung stimmt, das musst du noch selber errechnen aber diese Blockly's stehen ja oben. Wie gesagt für Akku berechne ich das nicht. Wäre aber identisch den anderen Werten machbar, sofern du das haben möchtest.

**Comment by Tw1nh34d** [comment]
..eine Frage hätte ich noch. Du benutzt in der Monatsbrechnung (Blockly) "PV generation total". Diese ID finde ich bei mir nicht. 
Ich habe nur "PV Generation" gefunden, siehe Screenshot:

![grafik](https://user-images.githubusercontent.com/55540912/117422084-83804280-af1f-11eb-966a-00c1e289cfe4.png)

**Comment by nobl** [comment]
Ich mach das hier mal nochmal auf :-) dann findet ihr diskutanten das Thema schneller wieder.

Da "Monat" ja offensichtlich ein echt heißer Kandidat ist: Ich überleg mir was, das systemseitig einzubauen ... Kein Versprechen - bin gerade etwas "Land unter" mit der Zeit.

**Comment by nobl** [comment]
Ich habe eben auf github aktualisiert.
Das Problem mit "ich muss mir Tages/Wochen/Monats/Jahres-Werte selbst berechnen sollte damit Geschichte sein".

**Comment by Tw1nh34d** [comment]
> 
> 
> Ich habe eben auf github aktualisiert.
> Das Problem mit "ich muss mir Tages/Wochen/Monats/Jahres-Werte selbst berechnen sollte damit Geschichte sein".

Hi nobl,

jetzt sind  ja wirklich alle Daten vorhanden. Top - Vielen Dank für die schnelle Umsetzung!!!

Grüße

**Comment by Tw1nh34d** [comment]
@nobl 

![grafik](https://user-images.githubusercontent.com/55540912/117704377-1395e480-b1cb-11eb-8396-98084880be5d.png)

Kann es sein, dass die Namen vertauscht sind:
Der Wert von gestern ist höher alls der Wochen- bzw. Jahreswert. Das kann eigentlich nicht sein, oder ich mache ich einen Denkfehler :)

**Comment by nobl** [comment]
doch das stimmt schon :-)
vor einem erstmaligen wochen/monats/jahreswechsel sind die werte mehr oder weniger "falsch", da es keinen korrekten referenzwert gibt.

du könntest auf mein-senec.de nachsehen, was dein theoretisch korrekter wert für jahr und monat wäre und auf dieser basis den jeweils zugehörigen referenzwert von hand korrigieren (diese werte kann man von hand anpassen).

dein referenzwert für den jahreswert ist offensichtlich höher als der für den tageswert.

Da die neue Woche Montag beginnt, kannst du Sonntag durchaus einen höheren Tageswert (yesterday) haben, als in der Woche.

allerdings wundert mich insgesamt, warum bei dir tag und woche unterschiedlich sind. An einem Montag müssten die parallel laufen (was sie bei mir auch tun).

Schau dir mal die Referenzwerte an, ob da was komisch ist.

(Alternativ: Adapter stoppen, den _calc Pfad einmal löschen und Adapter neu starten. Dann sollten alle Referenzwerte erstmal ident. sein und ab da verfolgen, was passiert. Das wäre auch ein guter Zeitpunkt, um die Referenzwerte dann anhand mein-senec anzupassen, sprich um die benötigten kWh zu reduzieren.)

Theoretisch kann sowas eigentl. nur vorkommen, wenn manuell einer der Werte oder der Ref-Werte manipuliert wurde.

Edit:
Eine Erklärung hätte ich allerdings doch noch: Falls du eine der ersten Versionen hattest, die nur den Tageswert mitgeführt hat, dann wäre der rest "klar", denn woche, monat, jahr habe ich erst später hinzugefügt, womit auch der gleichstand bei woche/jahr sich erklären würde. Beim Update wären die Referenzwerte für Woche / Jahr (und auch Monat) erstmalig gesetzt worden und damit höher gewesen, als der Referenzwert für den aktuellen Tag.

daher mein Vorschlag wie gehabt:
Alle Einträge weg, neu anlegen lassen, dann die Referenzwerte entspr. mein-senec.de anpassen.
(Geht auch anderweitig, allerdings müsstest du dann rechnen: referenzwert_neu = referenzwert_aktuell - SollWert aus meinSenec + aktueller_wert)

**Comment by Tw1nh34d** [comment]
> 
> 
> doch das stimmt schon :-)
> vor einem erstmaligen wochen/monats/jahreswechsel sind die werte mehr oder weniger "falsch", da es keinen korrekten referenzwert gibt.
> 
> du könntest auf mein-senec.de nachsehen, was dein theoretisch korrekter wert für jahr und monat wäre und auf dieser basis den jeweils zugehörigen referenzwert von hand korrigieren (diese werte kann man von hand anpassen).
> 
> dein referenzwert für den jahreswert ist offensichtlich höher als der für den tageswert.
> 
> Da die neue Woche Montag beginnt, kannst du Sonntag durchaus einen höheren Tageswert (yesterday) haben, als in der Woche.
> 
> allerdings wundert mich insgesamt, warum bei dir tag und woche unterschiedlich sind. An einem Montag müssten die parallel laufen (was sie bei mir auch tun).
> 
> Schau dir mal die Referenzwerte an, ob da was komisch ist.
> 
> (Alternativ: Adapter stoppen, den _calc Pfad einmal löschen und Adapter neu starten. Dann sollten alle Referenzwerte erstmal ident. sein und ab da verfolgen, was passiert. Das wäre auch ein guter Zeitpunkt, um die Referenzwerte dann anhand mein-senec anzupassen, sprich um die benötigten kWh zu reduzieren.)
> 
> Theoretisch kann sowas eigentl. nur vorkommen, wenn manuell einer der Werte oder der Ref-Werte manipuliert wurde.
> 
> Edit:
> Eine Erklärung hätte ich allerdings doch noch: Falls du eine der ersten Versionen hattest, die nur den Tageswert mitgeführt hat, dann wäre der rest "klar", denn woche, monat, jahr habe ich erst später hinzugefügt, womit auch der gleichstand bei woche/jahr sich erklären würde. Beim Update wären die Referenzwerte für Woche / Jahr (und auch Monat) erstmalig gesetzt worden und damit höher gewesen, als der Referenzwert für den aktuellen Tag.
> 
> daher mein Vorschlag wie gehabt:
> Alle Einträge weg, neu anlegen lassen, dann die Referenzwerte entspr. mein-senec.de anpassen.
> (Geht auch anderweitig, allerdings müsstest du dann rechnen: referenzwert_neu = referenzwert_aktuell - SollWert aus meinSenec + aktueller_wert)

Ich habe den Pfad _calc einmal gelöscht. Jetzt sind die Werte erstmal synchron. Die Refereznzwerte stimmen mit denen auf "mein-senec" überein.
Ich werde nun beobachten! :)

**Comment by noffycws** [comment]
Das mit den kalkulierten Daten finde ich ne super Sache. Danke dafür.

Könnte man das noch ergänzen, um auch die Autarkie (Jetzt/Tag/Woche/Monat/Jahr + je 1 davor) in Prozent zu berechnen? Wobei die Autarkie der Wert ist, den ich selbst erzeuge ohne etwas aus dem Netz zu ziehen.

Mein Vorschlag für die Formel (am Beispiel der Datenpunkte aus "_calc"):

`Autarkie (in %) = ((LIVE_PV_GEN - LIVE_GRID_EXPORT - LIVE_BAT_CHARGE_MASTER + LIVE_BAT_DISCHARGE_MASTER) / LIVE_HOUSE_CON) * 100`

Abhängig von den "Rundungsfehlern" der _calc-Werte passt das ungefähr zu den Daten, die ich im meine-senec-Portal habe.

Was haltet ihr davon?

**Comment by nobl** [comment]
added in @version 1.3.1

**Comment by nobl** [comment]
Ich schließe das hier jetzt mal nach bald einem Jahr.
Es ist ja nach wie vor möglich, bei aufkommenden Fragen darauf zu referenzieren.
